src/ripple/app/tx/impl/NFTokenBurn.cpp

```diff
@@ -104,9 +109,35 @@ NFTokenBurn::doApply()
         view().update(issuer);
     }
 
-    // Optimized deletion of all offers.
-    nft::removeAllTokenOffers(view(), keylet::nft_sells(ctx_.tx[sfNFTokenID]));
-    nft::removeAllTokenOffers(view(), keylet::nft_buys(ctx_.tx[sfNFTokenID]));
+    if (ctx_.view().rules().enabled(fixUnburnableNFToken))
+    {
+        // Delete up to 500 offers, but prefers buy offers first
+        int const deletedBuyOffers = nft::removeTokenOffersWithLimit(
+            view(),
+            keylet::nft_buys(ctx_.tx[sfNFTokenID]),
+            maxDeletableTokenOfferEntries);
+
+        if (maxDeletableTokenOfferEntries - deletedBuyOffers > 0)
+        {
+            nft::removeTokenOffersWithLimit(
+                view(),
+                keylet::nft_sells(ctx_.tx[sfNFTokenID]),
+                maxDeletableTokenOfferEntries - deletedBuyOffers);
+        }
+    }
+    else
+    {
+        // Deletion of all offers.
+        nft::removeTokenOffersWithLimit(
+            view(),
+            keylet::nft_sells(ctx_.tx[sfNFTokenID]),
+            std::numeric_limits<int>::max());
+
+        nft::removeTokenOffersWithLimit(
+            view(),
+            keylet::nft_buys(ctx_.tx[sfNFTokenID]),
+            std::numeric_limits<int>::max());
```

Comment:
Just FYI, if we were not removing _all_ of the NFT offers while we're inside this `else`, then this would be a transaction breaking change.  However since we're removing every last one of the NFT offers, it doesn't matter what order we delete them in.  The order of deletion does not affect the resulting metadata.

There's nothing that needs to be changed here.  I'm leaving a note in case you work on other transactors in the future.